### PR TITLE
Bug fix hide application dropdown when no application in namespace

### PR DESCRIPTION
**Pull request by vikram-raj**
Bug fix - https://jira.coreos.com/browse/ODC-1700

Peek 2019-08-31 04-22

### CODE REVIEW

File: frontend/packages/dev-console/src/components/import/app/ApplicationSelector.tsx [hunk 1]
```diff
@@ -48,7 +48,6 @@ const ApplicationSelector: React.FC<ApplicationSelectorProps> = ({ namespace })
           label="Application"
           helperTextInvalid={errorMessage}
           isValid={isValid}
-          isRequired
```
Comment (vikram-raj) [inline frontend/packages/dev-console/src/components/import/app/ApplicationSelector.tsx]: Missed this change in this PR https://github.com/openshift/console/pull/2550#issuecomment-526789480

File: frontend/packages/dev-console/src/components/import/import-validation-utils.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export const validationSchema = yup.object().shape({
   }),
   application: yup.object().shape({
     name: yup.string(),
-    selectedKey: yup.string().required('Required'),
```
Comment (vikram-raj) [inline frontend/packages/dev-console/src/components/import/import-validation-utils.ts]: Missed this change in this PR https://github.com/openshift/console/pull/2550#issuecomment-526789480

File: frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx [hunk 3]
```diff
@@ -135,6 +136,10 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
     ) {
       this.onChange(allSelectorItem.allSelectorKey);
     }
+
+    if (_.isEmpty(sortedList) && this.props.onLoad) {
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx]: This should only be called when `this.props.loaded === true`, no?
Also, deconstruct `onLoad` from `props` at the start of this function.

File: frontend/packages/dev-console/src/components/import/app/ApplicationSelector.tsx [hunk 4]
```diff
@@ -30,29 +32,39 @@ const ApplicationSelector: React.FC<ApplicationSelectorProps> = ({ namespace })
     validateForm();
   };
 
+  const handleOnLoad = (items: { [key: string]: string }) => {
+    setIsZeroApplication(_.isEmpty(items));
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/import/app/ApplicationSelector.tsx]: Should store the result of `_.isEmpty(items)` in a local var to reference on the next line.

File: frontend/packages/dev-console/src/components/import/app/ApplicationSelector.tsx [hunk 5]
```diff
@@ -30,29 +32,39 @@ const ApplicationSelector: React.FC<ApplicationSelectorProps> = ({ namespace })
...
+            id={fieldId}
+            namespace={namespace}
+            actionItem={{
+              actionTitle: 'Create New Application',
+              actionKey: CREATE_APPLICATION_KEY,
+            }}
+            autoSelect
+            selectedKey={selectedKey.value}
+            onChange={onDropdownChange}
+            onLoad={(items) => handleOnLoad(items)}
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/import/app/ApplicationSelector.tsx]:
```suggestion
            onLoad={handleOnLoad}
```

File: frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx (hunk 3 above)
Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx]: Yes, that is already checked on line no. 78.

Comment (christianvogt) [reply frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx]: What about only on first load?
`if (nextProps.loaded && !this.props.loaded)`

Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx]: Agree. 
Done.

File: frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx [hunk 6]
```diff
@@ -137,6 +140,10 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
           : _.get(_.keys(sortedList), 0);
     }
     selectedItem && this.handleChange(selectedItem, sortedList);
+
+    if (nextProps.loaded && !this.props.loaded && _.isEmpty(sortedList) && onLoad) {
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx]: You don't need to check if the list is empty here. Always notify.

Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx]: Yes, no need to check for an empty list as now I am sending the list item.

Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx]: Done. Remove the check for the empty list.

File: frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx [hunk 7]
```diff
@@ -72,6 +73,8 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
       autoSelect,
     } = nextProps;
 
+    const { onLoad } = this.props;
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx]: `onLoad` should be coming from `nextProps`

**Comment by andrewballantyne** [comment]
Other than the `onLoad` being moved to nextProps

/lgtm

File: frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx (hunk 7 above)
Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx]: Done.

**Pull request merged**